usr/local/share/bastille/etcupdate.sh

```diff
@@ -0,0 +1,254 @@
+#!/bin/sh
+#
+# Copyright (c) 2018-2023, Rodrigo Nascimento Hernandez <rodrigomdev@gmail.com>
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+. /usr/local/share/bastille/common.sh
+. /usr/local/etc/bastille/bastille.conf
+
+usage() {
+        # Update /etc folder while keeping user changes
+        error_notify "Usage: bastille etcupdate [option(s)(optional)] [jailname] [oldrelease] [newrelease]"
+
+        cat << EOF
+        Options:
+
+        -D | --dryrun   -- Do a dry run. Output actions to stdout but without making changes.
+        -Q | --quiet    -- Do not output actions to stdout
+
+EOF
+        exit 1
+}
+
+
+executeconditional() {
+    if [ $DRY_RUN -eq "0" ]; then
+        eval "$@"
+    fi
+}
+
+C1_C6_conditions() {
+    filelistjail=$(find "${jail_etc}" -mindepth 1 -type f)
+    for jailfile in ${filelistjail}
+    do
+        filepart=$(echo "${jailfile}" | awk -F 'etc/' '{print $NF}')
+        newbasefile="${new_basedir}/etc/${filepart}"
+        currentbasefile="${current_basedir}/etc/${filepart}"
+
+        if [ ! -f "${currentbasefile}" ]; then
+            if [ ! -f "${newbasefile}" ]; then
+                C1=$((C1+1))
+                C1ct="$C1ct${jailfile}\n"
+            else
+                C2=$((C2+1))
+                C2ct="$C2ct${jailfile}\n"
+            fi
+        fi
+
+        if [ -f "${currentbasefile}" ]; then
+            diffr=$(diff -u "${jailfile}" "${currentbasefile}")
```

Comment:
Consider using temp files rather than variables here.  Variables can have unexpected whitespace folding and length issues, which might lead to unexpected corruption.  

Also, if [`diff3`](https://man.freebsd.org/cgi/man.cgi?query=diff3&sektion=1&apropos=0&manpath=freebsd) is available (i.e. `if command -v diff3 >/dev/null 1>&2; then ... fi`), perhaps use a threeway diff to generate your updated files, e.g. `diff3 -A "${newbasefile}" "${currentbasefile}" "${jailfile}"`.  This should be able to better differentiate between changes that were made by the end-user and changes that were introduced by the update.

Perhaps use `diffdir="$(mktemp -d -t "${TMPDIR:-/tmp}/etcupdate.XXXXXX")"` to make a temporary directory, then write your diffs into the temp dir.  Let the end-user decide whether to use your version or the existing version, then move the desired version out of the directory into the correct place.  Use `trap "[ -d '${diffdir}' ] && rm -rf -- '${diffdir}'" EXIT INT TERM` to clean up all unused temp files at the end of the run.

Perhaps allow end-user to invoke `vi` (or `"${VISUAL:-${EDITOR:-vi}}"` for style points) on the updated file, so they can make manual edits to each file prior to acceptance.  Especially, if doing in-place diffs with `diff3`, you can use the exit code of `diff3` to determine if a diff conflict has happened (`[ "$?" -eq 1 ]`), and prompt them to manually resolve.